bundles/org.eclipse.e4.ui.workbench.renderers.swt/src/org/eclipse/e4/ui/workbench/renderers/swt/MenuManagerRenderer.java

```diff
@@ -1170,7 +1172,16 @@ private void scheduleManagerUpdate(IContributionManager mgr) {
 							mgrToUpdate.clear();
 						}
 						for (IContributionManager mgr1 : toUpdate) {
-							mgr1.update(false);
+							try {
+								mgr1.update(false);
+							} catch (RuntimeException e) {
+								String message = String.format(
+										"ContributionManager '%s' threw an exception while performing update!", mgr1); //$NON-NLS-1$
+								ILog.get().error(message, e);
+								synchronized (mgrToUpdate) {
+									mgrToUpdate.add(mgr1);
```

Comment:
Wouldn't this spam the log in case of a programming error? Our code, for example, would throw an exception every time you ran it.

Reply:
It will only be ran again if something changed what would then produce a log message probably again. I just don't see a good way to prevent this as the manager itself might not be responsible for that directly it seems a bit hard to suppress any executions (it might only be a temporary problem).